src/groupby/hash/groupby_compute_api.h

```diff
@@ -0,0 +1,210 @@
+                        aggregation_operation aggregation_op,
+                        bool sort_result = false)
+{
+
+  using map_type = concurrent_unordered_map<groupby_type, 
+                                            aggregation_type, 
+                                            std::numeric_limits<groupby_type>::max(), 
+                                            default_hash<groupby_type>, 
+                                            equal_to<groupby_type>, 
+                                            legacy_allocator<thrust::pair<groupby_type, aggregation_type> > >;
+
+  cudaError_t error{cudaSuccess};
+
+  // Inputs cannot be null
+  if(in_groupby_column == nullptr || in_aggregation_column == nullptr)
+    return cudaErrorNotPermitted;
+
+  // Input size cannot be 0 or negative
+  if(in_column_size <= 0)
+    return cudaErrorNotPermitted;
+
+  // Output buffers must already be allocated
+  if(out_groupby_column == nullptr || out_aggregation_column == nullptr)
+    return cudaErrorNotPermitted;
+
+  std::unique_ptr<map_type> the_map;
+
+  // The hash table occupancy and the input size determines the size of the hash table
+  // e.g., for a 50% occupancy, the size of the hash table is twice that of the input
+  const size_type hash_table_size = static_cast<size_type>((static_cast<size_t>(in_column_size) * 100 / DEFAULT_HASH_TABLE_OCCUPANCY));
+
+  const dim3 build_grid_size ((in_column_size + THREAD_BLOCK_SIZE - 1) / THREAD_BLOCK_SIZE, 1, 1);
+  const dim3 block_size (THREAD_BLOCK_SIZE, 1, 1);
+
+  // Initialize the hash table with the aggregation operation functor's identity value
+  the_map.reset(new map_type(hash_table_size, aggregation_operation::IDENTITY));
+
+  //the_map->prefetch(0);
+
+  CUDA_RT_CALL(cudaGetLastError());
+
+  // Inserts (groupby_column[i], aggregation_column[i]) as a key-value pair into the
+  // hash table. When a given key already exists in the table, the aggregation operation
+  // is computed between the new and existing value, and the result is stored back.
+  build_aggregation_table<<<build_grid_size, block_size>>>(the_map.get(), 
+                                                           in_groupby_column, 
+                                                           in_aggregation_column,
+                                                           in_column_size,
+                                                           aggregation_op);
+  CUDA_RT_CALL(cudaGetLastError());
+
+  // Used by threads to coordinate where to write their results
+  unsigned int * global_write_index{nullptr};
+  CUDA_RT_CALL(cudaMallocManaged(&global_write_index, sizeof(unsigned int)));
+  CUDA_RT_CALL(cudaMemset(global_write_index, 0, sizeof(unsigned int)));
+
+  CUDA_RT_CALL(cudaDeviceSynchronize());
+
+  const dim3 extract_grid_size ((the_map->size() + THREAD_BLOCK_SIZE - 1) / THREAD_BLOCK_SIZE, 1, 1);
+
+  // Extracts every non-empty key and value into separate contiguous arrays,
```

Comment:
Does this comment apply to the following dead code? Are you keeping it in so we could see how it WOULD have been called without the kernel failing to launch issue?

Reply:
The comment applies to both the <<< >>> kernel launch as well as the cudaLaunchKernel. The same kernel is being run, and therefore the result is identical, but for some reason the <<< >>> syntax would fail. I could remove the <<< >>> kernel call, but it is a bug I need to isolate and look more into in the future.